src/backend/core/api/viewsets.py

```diff
@@ -494,6 +508,44 @@ def link_configuration(self, request, *args, **kwargs):
         serializer.save()
         return drf_response.Response(serializer.data, status=status.HTTP_200_OK)
 
+    @decorators.action(detail=True, methods=["post", "delete"], url_path="favorite")
+    def favorite(self, request, *args, **kwargs):
+        """
+        Mark or unmark the document as a favorite for the logged-in user based on the HTTP method.
+        """
+        # Check permissions first
+        document = self.get_object()
+        user = request.user
+
+        if request.method == "POST":
+            # Try to mark as favorite
+            _, created = models.DocumentFavorite.objects.get_or_create(
+                document=document, user=user
+            )
```

Comment:
Nit: For performance, and because you don't need to actually get the data from the database, I would rather do directly a try/except to create the object.

Reply:
agreed!